Java Data Structures Review (#316)
NuevoFoundation/workshops: content/english/java-datastructures/arraylists/problem-1.md

```js javascript
ArrayList<String> menu = new ArrayList<>(); 

menu.add("Pizza"); 
menu.add("Hotdog"); 
menu.add("Hamburger"); 
menu.add("Hotdog"); 

// If we run the code
item = find(menu, 0);

// find() will return since it is the 0th item in the menu
item = "Pizza"
```

{{%notice tip%}}
1. How can you iterate through the list?
2. Look at the example menu!
{{%/notice%}}

<iframe frameborder="0" width="100%" height="800px" src="https://replit.com/@nuevofoundation/2DFind?lite=true"></iframe>

## Task 3: Removing Elements

The same Chinese restaurant has a bug in their code! Orders have been duplicated randomly and the `ArrayList` are filled with copies of orders. Help the owner out by writing a program to remove the first `n` occurrences of a given order in the `ArrayList`.

```js javascript
ArrayList<String> menu = new ArrayList<>(); 

menu.add("Pizza"); 
menu.add("Hotdog"); 
menu.add("Hamburger"); 
menu.add("Pizza");
menu.add("Pizza");
menu.add("Hotdog"); 

> // If we run the code
> item = remove(menu,"Pizza", 2);
> 
> // remove() will remove the first two Pizza orders
> menu = {"Hotdog", "Hamburger", "Pizza", "Hotdog"}

Reviewer: ```suggestion
// Remove the first two Pizza orders
item = remove(menu,"Pizza", 2);

// menu will now be the following array: {"Hotdog", "Hamburger", "Pizza", "Hotdog"}
```
